src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java

```diff
@@ -65,39 +75,81 @@ public GetTasksResponse<Task> searchForTasks(
-
-        //TODO this should pass only prepared data.
-        // Role Assignments that are filtered and grouped already.
-        // Should also be in one object called SearchData
-        //SearchData searchData = new TaskSearchData(
-        //    new AndPermissionsRequired(permissionsRequired),
-        //    Lists.newArrayList(),//TODO put RoleAssignmentForSearch list in there
-        //    searchTaskRequest);
-        //
-        //final Specification<TaskResource> taskQuerySpecification = TaskSearchQueryBuilder.build(searchData);
-
-        final Specification<TaskResource> taskResourceSpecification =
-            TaskSearchQueryBuilder.buildTaskQuery(
-                searchTaskRequest,
-                accessControlResponse,
-                permissionsRequired);
+        final List<TaskResourceSummary> taskResourcesSummary = getTaskResourceSummary(
+            firstResult,
+            maxResults,
+            searchTaskRequest,
+            accessControlResponse,
+            permissionsRequired
+        );
 
+        if (isEmpty(taskResourcesSummary)) {
+            return new GetTasksResponse<>(List.of(), 0);
+        }
 
-        final Page<TaskResource> pages = taskResourceRepository.findAll(taskResourceSpecification, page);
+        final List<TaskResource> taskResources = getTaskResources(searchTaskRequest, taskResourcesSummary);
 
-        final List<TaskResource> taskResources = pages.toList();
+        Long count = getTotalCount(searchTaskRequest, accessControlResponse, permissionsRequired);
 
         final List<Task> tasks = taskResources.stream()
             .map(taskResource ->
-                cftTaskMapper.mapToTaskAndExtractPermissionsUnion(
-                    taskResource,
-                    accessControlResponse.getRoleAssignments())
+                     cftTaskMapper.mapToTaskAndExtractPermissionsUnion(
+                         taskResource,
+                         accessControlResponse.getRoleAssignments()
+                     )
             )
             .collect(Collectors.toList());
 
-        return new GetTasksResponse<>(tasks, pages.getTotalElements());
+        return new GetTasksResponse<>(tasks, count);
+    }
+
+    private List<TaskResourceSummary> getTaskResourceSummary(int firstResult,
+                                                             int maxResults,
+                                                             SearchTaskRequest searchTaskRequest,
+                                                             AccessControlResponse accessControlResponse,
+                                                             List<PermissionTypes> permissionsRequired) {
+        Sort sort = SortQuery.sortByFields(searchTaskRequest);
+        Pageable page = OffsetPageableRequest.of(firstResult, maxResults, sort);
+        TaskResourceSummaryQueryBuilder summaryQueryBuilder = new TaskResourceSummaryQueryBuilder(entityManager);
```

Comment:
If all we do is pass the entityManager elsewhere do we need it in here or is it better to have it injected into where we are passing it?

Reply:
I think it's ok for now as I think we most likly want to reorganise/refactor later anyway.

Reply:
I have started refactoring in another PR